docs/modules/ROOT/pages/documentation/microprofile/config/cloud/azure.adoc

```diff
@@ -0,0 +1,116 @@
+
+## Add Credentials
+
+Credentials allow your application to authenticate as itself. Currently Azure supports two options as credentials: client secrets (a string) and certificates (public key). Currently, Payara only supports certificate as a credential as it provides a higher level of assurance than a client secret.
+
+To upload for own certificate for authentication. Select your application, under Manage section select Certificates & secrets, select Upload certificate, select the certificate you would like to upload and select Add. Once the certificate is successfully uploaded it should generate a Thumbprint, this will be required when we configure the Azure Config Source.
+
+image:microprofile/config/cloud/azure/certificate-upload.png[Certificate Upload]
+
+## Create a Key Vault
+
+To securely store your secrets you will need to create a key vault. To create a key valut using the Azure portal, search for key vaults and select it, select Add and provide relvant information on the Create key vault page and finally select Review + create.  
+
+NOTE: Although key vault allows storing of keys, secrets, and certificates, Azure Config Source currently only support secrets.
+
+image:microprofile/config/cloud/azure/key-vault-creation.png[Key Vault Creation]
+
+You can see from the above diagram I have only configured the Basic section and other section such as Access policy and Networking has been left to their defaults. You can confguire these section yourslef to fit your requirments.
+
+### Configure access policies on Key Vault
+
+To give an application access to secrets in a key vault you will need to add the application to the key vault's acess policies. To do this using a Azure portal, navigate to your key vault, select Access policies, select Add Access Policy, under Add access policy page for *Configure from template* select Secret Management as we don't require others, select the service principal you created previously, then select Add and finally select Save to commit your changes. 
+
+image:microprofile/config/cloud/azure/access-policy-configuration.png[Access Policy Configuration]
+
+### Add a Secret to Key Vault
+
+To add a secert to the key vault, select the key vault you want to add your secert, under Settings section select Secrets, then select Generate/Import, provide relvant information on the Create a secret page and finally select Create. 
+
+image:microprofile/config/cloud/azure/secret-creation.png[Secret Creation]
+
+[[configuration]]
+## Configuration
+
+You can configure Azure Secrets either via the admin console or the _asadmin_ utility. You'll need the key vault name you created in the pervious section, tenant ID and client ID of you application you crerated in the pervious section, your private key file and thumbprint generated by your application after you uploaded your certificate.  
+
+NOTE: Your private key file will be copied into the Payara Server config/ directory.
```

Comment:
If you are referring to the domain's configuration directory, better refer to it as such or by the `${PAYARA_DOMAIN}/config` path instead